### PR TITLE
test(Lead): add LeadSaveControllerTest to test save correct urls

### PR DESCRIPTION
**Pull request by elguitarraverde**
Right now the tests are failing because currently the blank spaces are saved in the database.

consider using:
```php
$name = $request->string('linkedin')->trim();
```
https://laravel.com/docs/9.x/requests#retrieving-stringable-input-values
https://laravel.com/docs/9.x/requests#determining-if-input-is-present
https://laravel.com/docs/9.x/requests#input-trimming-and-normalization

https://laravel.com/docs/9.x/eloquent-mutators#defining-a-mutator

https://github.com/Roskus/hammercrm/blob/b19f2afb4d8ba3cf57365b931bd8bc5156f9f4d8/app/Http/Controllers/Lead/LeadSaveController.php#L38-L43

### CODE REVIEW

**Comment by gnovaro** [comment]
I needed to add this validation after the strict_types add .. I agree with you we can do it better

**Comment by elguitarraverde** [comment]
I have configured my IDE so that from now on every file I add declare(strict_types=1).

**Review by gnovaro — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Lead/LeadSaveController.php [hunk 1]
```diff
@@ -38,9 +38,9 @@ public function save(Request $request)
         $lead->linkedin = ($request->linkedin) ? rtrim($request->linkedin, '/') : null;
         $lead->facebook = ($request->facebook) ? rtrim($request->facebook, '/') : null;
         $lead->instagram = ($request->instagram) ? rtrim($request->instagram, '/') : null;
-        $lead->twitter = ($request->twitter) ? rtrim($request->twitter, '/'): null;
-        $lead->youtube = ($request->youtube) ? rtrim($request->youtube, '/'): null;
-        $lead->tiktok = ($request->tiktok) ? rtrim($request->tiktok, '/'): null;
+        $lead->twitter = ($request->twitter) ? rtrim($request->twitter, '/') : null;
```
[the remark this answers is not in the document]
Comment (elguitarraverde) [reply app/Http/Controllers/Lead/LeadSaveController.php]: Only with this, the tests pass with the three cases that I initially programmed. Take a look and add casuistry that may occur.

```php
$lead->website = Str::of($request->website)->trim()->beforeLast('?');
```

you car run: `vendor/bin/phpunit --filter it_can_save_urls`

https://github.com/Roskus/hammercrm/blob/b1a5222cf094c7fb935da9f5f35fb48199590155/tests/Unit/Controllers/Lead/LeadSaveControllerTest.php#L21-L34

File: tests/Unit/Controllers/Lead/LeadSaveControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+class LeadSaveControllerTest extends TestCase
+{
+    /** @test */
+    public function it_can_save_urls(): void
+    {
+        $user = User::factory()->create();
+        $this->actingAs($user);
+
+        $urls = [
```
Comment (gnovaro) [inline tests/Unit/Controllers/Lead/LeadSaveControllerTest.php:21]: There is another way to do this with @dataProviders 

https://blog.martinhujer.cz/how-to-use-data-providers-in-phpunit/

**Review by gnovaro — CHANGES_REQUESTED** [review]
Looks great but just a suggestion is use dataProviders

File: tests/Unit/Controllers/Lead/LeadSaveControllerTest.php (hunk 2 above)
Comment (elguitarraverde) [reply tests/Unit/Controllers/Lead/LeadSaveControllerTest.php]: https://github.com/Roskus/prospect-flow-crm/pull/80/commits/1b0f7c63d67226331740305d2e10b6a208fc766f

[add Data Providers to Lead Save urls](https://github.com/Roskus/prospect-flow-crm/pull/80/commits/1b0f7c63d67226331740305d2e10b6a208fc766f)

**Review by gnovaro — APPROVED** [review]
Thanks for the improvement looks better! Great job

**Review by gnovaro — APPROVED** [review]

**Comment by elguitarraverde** [comment]
We will return to the topic later.

**Pull request closed without merging**